arches/app/utils/compatibility.py

```diff
@@ -0,0 +1,40 @@
+import logging
+from arches import VERSION
+from arches.app.models.system_settings import settings
+from django.utils.translation import gettext as _
+
+logger = logging.getLogger(__name__)
+
+
+def is_arches_compatible(min_arches=settings.MIN_ARCHES_VERSION, max_arches=settings.MAX_ARCHES_VERSION, target="project"):
+    """
+    Check if the current version of arches falls between a min and max version.
+
+    Keyword arguments
+    min_arches -- A tuple with at least a major version of arches, but may also include a min and patch release e.g. (7,4,0)
+    max_arches -- A tuple with at least a major version of arches, but may also include a min and patch release e.g. (7,5,0)
+    target -- A description of what is being checked for compatibility
+
+    """
+
+    if min_arches is None or max_arches is None:
+        logger.error(
+            _("Either a minimum or maximum compatible Arches version is not specified. Unable to check {0} compatibility".format(target))
+        )
+        return
+
+    if len(min_arches) <= 3 and len(max_arches) <= 3:
```

Comment:
I'm not sure that we should **require** that a project creator specify both a min and max compatible version, or either for that matter.  A required min version might make sense, but a max version is harder to understand.  
At the moment packages aren't pip installed, so upgrading to keep packages and arches versions in sync is more difficult.  

Was any thought put to using a library like [this](https://github.com/python-semver/python-semver) to manage version numbers and comparisons?

Reply:
@chiatt Yeah, that all makes sense to me.